src/main/java/com/aws/greengrass/easysetup/GreengrassSetup.java

```diff
@@ -429,6 +429,11 @@ void provision(Kernel kernel) throws IOException, DeviceConfigurationException {
                     kernel.getContext().get(DeviceConfiguration.class).getNucleusVersion());
         }
 
+        // Dump config since we've just provisioned so that the bootstrap config will enable us to
+        // reach the cloud when needed. Must do this now because we normally would never overwrite the bootstrap
+        // file, however we need to do it since we've only just learned about our endpoints, certs, etc.
+        kernel.writeEffectiveConfigAsTransactionLog(kernel.getNucleusPaths().configPath()
```

Comment:
Should this go into Kernel./KernelLifecycle so that even if the config crucial for connecting to cloud changes via deployment, we will have the right set to fallback to on restarting after a crash?

Reply:
No, we decided that we never want to write to the file, at least initially. So we will not update it in response to a deployment.

Reply:
So if the config does get changed through deployment then this issue will still be a problem right? Because it's possible that the bootstrap version wouldn't work for connecting the device to cloud anymore

Reply:
Potentially yes, we did think about that and we're ok with it for now.